pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -394,11 +408,43 @@ base mixin DartToolingDaemonSupport on ToolsSupport {
               ),
             ],
           );
-        } finally {
-          await vmService.callServiceExtension(
-            '$inspectorExtensionPrefix.disposeGroup',
+        }
+      },
+    );
+  }
+
+  /// Retrieves the selected widget from the currently running app.
+  ///
+  /// If more than one debug session is active, then it just uses the first one.
+  // TODO: support passing a debug session id when there is more than one debug
+  // session.
+  Future<CallToolResult> selectedWidget(CallToolRequest request) async {
+    return _callOnVmService(
+      callback: (vmService) async {
+        final vm = await vmService.getVM();
+        final isolateId = vm.isolates!.first.id;
+        try {
+          final result = await vmService.callServiceExtension(
+            '$_inspectorServiceExtensionPrefix.getSelectedSummaryWidget',
             isolateId: isolateId,
-            args: {'objectGroup': groupId},
+            args: {'objectGroup': inspectorObjectGroup},
+          );
+
+          final widget = result.json?['result'];
+          if (widget == null) {
+            return CallToolResult(
+              content: [
+                TextContent(text: 'No Widget selected.'),
+              ],
+            );
+          }
+          return CallToolResult(
+            content: [TextContent(text: widget.toString())],
```

Comment:
Should we do this instead?

```suggestion
            content: [TextContent(text: jsonEncode(widget)],
```

Or maybe even:

```suggestion
            content: [TextContent(text: JsonEncoder.withIndent(' ').encode(widget)],
```

Reply:
Will the added new lines caused by using .withIndent have any impact on the LLM trying to parse this information? In terms of using up more tokens than necessary?

Reply:
It probably is just a waste of tokens yes, no idea how it might affect its ability to reason about it either positively or negatively 🤷‍♂️ .

Just jsonEncode SGTM though